### PR TITLE
First draft of FDK guide

### PR DESCRIPTION
**Pull request by Khazakdk**
Non spec specific changes:

- Global rename of `getReducedCost` to `getAdjustedCost`
- Moved the data and spec inputs for ResourceGraph to getters than can be overridden (tagging @kfinch)

### CODE REVIEW

File: src/parser/classic/modules/resources/ManaTracker.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { CastEvent } from 'parser/core/Events';
 import ManaTracker from 'parser/core/healingEfficiency/ManaTracker';
 
 class TbcManaTracker extends ManaTracker {
-  getReducedCost(event: CastEvent) {
+  getAdjustCost(event: CastEvent) {
```
Comment (kfinch) [inline src/parser/classic/modules/resources/ManaTracker.ts]: Should all these be `getAdjustedCost`?

File: src/analysis/retail/deathknight/frost/Guide.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      {(info.combatant.hasTalent(talents.EMPOWER_RUNE_WEAPON_FROST_TALENT) ||
+        info.combatant.hasTalent(talents.EMPOWER_RUNE_WEAPON_SHARED_TALENT)) && (
+        <div className="flex-main chart" style={{ padding: 5 }}>
+          <CooldownBar
+            spellId={spells.EMPOWER_RUNE_WEAPON.id}
+            events={events}
+            info={info}
+            highlightGaps
+          />
+        </div>
```
Comment (kfinch) [inline src/analysis/retail/deathknight/frost/Guide.tsx]: I recently merged a refactor of `CooldownBar` that makes it require less stuff - you'll have to carry through the refactor to these usages for them to work once its merged. I recommend pulling from dragonflight.

The new usage will be: ```
<CooldownBar
  spellId={ ... }
  gapHighlightMode={GapHighlight.FullCooldown}
/>```

**Review by kfinch — COMMENTED** [review]
Can you post a screenshot of the Guide?

**Comment by Khazakdk** [comment]
image

**Review by kfinch — APPROVED** [review]

**Pull request merged**